src/Stratis.Bitcoin.IntegrationTests/EnvironmentMockUpHelpers/NodeRunners.cs

```diff
@@ -192,12 +267,99 @@ public static FullNode BuildFullNode(NodeSettings args, Action<IFullNodeBuilder>
 
             return node;
         }
+    }
+
+    public sealed class StratisProofOfWorkMiningNode : INodeRunner
+    {
+        private Action<IFullNodeBuilder> callback;
+        public FullNode FullNode { get; set; }
+        private bool mineCoinsFast;
+
+        public StratisProofOfWorkMiningNode(Action<IFullNodeBuilder> callback = null) : base()
+        {
+            this.callback = callback;
+        }
 
-        public FullNode FullNode;
+        public bool IsDisposed
+        {
+            get { return this.FullNode.State == FullNodeState.Disposed; }
+        }
+
+        public void Kill()
+        {
+            this.FullNode?.Dispose();
+        }
+
+        public void Start(string dataDir, bool mineCoinsFast)
+        {
+            NodeSettings nodeSettings = new NodeSettings(Network.StratisRegTest, ProtocolVersion.ALT_PROTOCOL_VERSION, args: new string[] { "-conf=stratis.conf", "-datadir=" + dataDir }, loadConfiguration: false);
+
+            this.mineCoinsFast = mineCoinsFast;
+
+            var node = BuildFullNode(nodeSettings, this.callback);
+
+            this.FullNode = node;
+            this.FullNode.Start();
+        }
+
+        public FullNode BuildFullNode(NodeSettings args, Action<IFullNodeBuilder> callback = null)
+        {
+            FullNode node;
+
+            if (callback != null)
+            {
+                var builder = new FullNodeBuilder().UseNodeSettings(args);
+                callback(builder);
+                node = (FullNode)builder.Build();
+            }
+            else
+            {
+                var builder = new FullNodeBuilder()
+                    .UseNodeSettings(args)
+                    .UsePosConsensus()
+                    .UseBlockStore()
+                    .UseMempool()
+                    .AddMining()
+                    .UseWallet()
+                    .AddRPC()
+                    .MockIBD();
+
+                if (this.mineCoinsFast)
+                    builder.SubstituteDateTimeProviderFor<MiningFeature>(new GenerateCoinsFastDateTimeProvider());
+
+                node = (FullNode)builder.Build();
+            }
+
+            return node;
+        }
     }
 
     public static class FullNodeTestBuilderExtension
```

Comment:
Hi, maybe move the extension class into it's own folder.  My [PR](https://github.com/stratisproject/StratisBitcoinFullNode/pull/1259) (which isn't merged to master yet) has an extension class - `CoreNodeExtension`.  

`CoreNodeExtension` lives in a \Utilities\Extensions folder.

![image](https://user-images.githubusercontent.com/16000683/38771457-8872d7f6-401b-11e8-991e-c7edadac32fb.png)

Reply:
If there are no other types of code in the \Utilities folder then maybe just an \Extensions folder.  I added Utilities as I copied an example from another project in the solution.

Reply:
OK Ive added this to a \Extensions folder.